docs/internal/runtime-upgrade.md

```diff
@@ -0,0 +1,208 @@
+# Runtime upgrade with the Runtime Provisioner
+
+## Upgrade Runtime
```

Comment:
Do we need this heading? I'd remove it and go on with the content.

Reply:
I think we do, cos there's another section on Rollback later.